src/App.tsx

```diff
@@ -1,61 +1,135 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState('0');
+  const [title, setTitle] = useState('');
+  const [todoList, setTodoList] = useState(todos);
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserEmpty, setIsUserEmpty] = useState(false);
+
+  const maxId = todoList.reduce((max, todo) => Math.max(max, todo.id), 0);
+
+  const handleUserSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(event.target.value);
+    if (isUserEmpty) {
+      setIsUserEmpty(false);
```

Comment:
I believe, it's better to perform this validation inside submit callback

Reply:
I tried, but I don't know how. If I move it to form handler it doesn't work like on screen(( it has to change immediately so I put it in the handler for input
![image](https://user-images.githubusercontent.com/87998640/234929735-f51be0cb-3978-4711-8b5e-bf781e9dfea0.png)